server/datastore/mysql/microsoft_mdm.go

```diff
@@ -63,8 +37,11 @@ func (ds *Datastore) MDMWindowsGetEnrolledDeviceWithDeviceID(ctx context.Context
 	return &winMDMDevice, nil
 }
 
-// MDMWindowsInsertEnrolledDevice inserts a new MDMWindowsEnrolledDevice in the database
+// MDMWindowsInsertEnrolledDevice inserts a new MDMWindowsEnrolledDevice in the
+// database.
 func (ds *Datastore) MDMWindowsInsertEnrolledDevice(ctx context.Context, device *fleet.MDMWindowsEnrolledDevice) error {
+	// TODO(mna): I think this needs to support an ON DUPLICATE UPDATE to
+	// handle the potential case of a device_id changing for a given hardware_id.
```

Comment:
I left a few TODO comments as notes of ways that I was planning on adjusting the code. If someone else takes over to move this forward while I'm away, feel free to use a different approach if you feel it fits better, that's just my current take on it and acts as some reminder of things to address.